code-mancers/dockup
# FakeDockup backend for better developer experience

Closes #115 

Uses "branch" name in deployment form to simulate various scenarios. Currently I have included only the happy path scenario with name "Scenario1" which can be used as the branch name in the deployment form when in dev env.

How to test:

1. Create a whitelisted git URL using link in navbar
2. Fill in the deployment form in homepage. Use "Scenario1" as the branch name
3. Deploy
4. Make sure you can see state changes in the UI.
5. Now try again with a random branch name, say, "foobar"
6. You should see a helpful message along with "Deployment failed" status change.

Note: I have changed this behaviour in a later commit (will open the PR once this is merged) a little bit. If the branch name can be matched with a scenario, it will be used, otherwise the default happy path will be chosen. ( Commit: https://github.com/code-mancers/dockup/commit/3e4f6acbb38c5ced48609ee80bd1e95fa0cdc636#diff-53ad342d6cc75041408c5e9f7f460423R10 )

If you are stuck at "Deploying", it could be because phoenix live reloading kicked in and screwed with the phoenix channel that gets live updates. Refresh the browser and try again if this happens.

PS: This PR also adds a `DockupSpec` behaviour which is to be implemented by a DockupUI compatible backend. This is currently used only for Dockup and FakeDockup. As part of this, added typespec to the specification. Not planning to fix all dialyzer warnings. Closes #105 .
